avocado/utils/distro.py

```diff
@@ -331,6 +331,40 @@ class FedoraProbe(RedHatProbe):
     CHECK_VERSION_REGEX = re.compile(r'Fedora release (\d{1,2}).*')
 
 
+class AmazonLinuxProbe(Probe):
+
+    """
+    Probe for Amazon Linux systems
+    """
+
+    CHECK_FILE = '/etc/os-release'
+    CHECK_FILE_CONTAINS = 'Amazon Linux'
+    CHECK_FILE_DISTRO_NAME = 'amzn'
+
+    def get_distro(self):
+        dst = Probe.get_distro(self)
+
+        version_re = re.compile(r'VERSION="(.*)"')
+        version = None
+
+        with open(self.CHECK_FILE) as check_file:
+            for line in check_file:
+                match = version_re.match(line)
+                if match:
+                    version = match.group(1)
+
+        if version:
+            version_parts = version.split()
+            if len(version_parts) == 1:
+                dst.version = 1
+                dst.release = re.findall(r'\d*\.\d*', version_parts[0])[0]
+            else:
+                dst.version = int(version_parts[0].split('.')[0])
+                dst.release = re.findall(r'\d*\.\d*', version_parts[1])[0]
```

Comment:
Actually this file exists on Fedora as well and reports `27 (Twenty Seven)`, which crashes with `IndexError` breaking the detection whatsoever. Would you please add a `NAME` check and handle possible IndexError gracefully?

Reply:
Btw you can inspire from SUSEProbe. Acutally you can perhaps just update it (to allow version_id to be split) and inherit from that one as it's very similar.